__tests__/app.test.js

```diff
@@ -155,4 +155,106 @@ describe("All requests", () => {
+                .expect(404)
+                .send({
+                    user_name: "butter_bridge",
+                    body: {
+                        article_id: 9,
+                        author: "butter_bridge",
+                        votes: 20,
+                        created_at: new Date(),
+                        body: "post request will going to be ok",
+                    },
+                })
+                .then(({ body }) => {
+                    const { msg } = body;
+                    expect(msg).toBe("Not found!");
+                });
+        });
+        test("404 should be returned when user_name from request body not exits in users table", () => {
+            return request(app)
+                .post("/api/articles/9/comments")
+                .expect(404)
+                .send({
+                    user_name: "David",
+                    body: {
+                        article_id: 9,
+                        author: "butter_bridge",
+                        votes: 20,
+                        created_at: new Date(),
+                        body: "post request will going to be ok",
+                    },
+                })
+                .then(({ body }) => {
+                    const { msg } = body;
+                    expect(msg).toBe("Not found!");
+                });
+        });
+        test("400 should be returned when the property article_id of the body cause foreign key relationship problem", () => {
+            return request(app)
+                .post("/api/articles/9/comments")
+                .expect(400)
+                .send({
+                    user_name: "David",
+                    body: {
+                        article_id: 999,
+                        author: "butter_bridge",
+                        votes: 20,
+                        created_at: new Date(),
+                        body: "post request will going to be ok",
+                    },
+                })
+                .then(({ body }) => {
+                    const { msg } = body;
+                    expect(msg).toBe("Invalid request!");
+                });
+        });
+        test("400 should be returned when the property author of the body cause foreign key relationship problem", () => {
+            return request(app)
+                .post("/api/articles/9/comments")
+                .expect(400)
+                .send({
+                    user_name: "David",
+                    body: {
+                        article_id: 9,
+                        author: "unKnown",
+                        votes: 20,
+                        created_at: new Date(),
+                        body: "post request will going to be ok",
+                    },
+                })
+                .then(({ body }) => {
+                    const { msg } = body;
+                    expect(msg).toBe("Invalid request!");
+                });
+        });
+    });
 });
```

Comment:
There are still some tests that should be added here.

What should we expect if the user made a request to an invalid article id?  e.g. /api/articles/banana/comments

What should we expect if the user does not give us the required fields? e.g. misisng username or body